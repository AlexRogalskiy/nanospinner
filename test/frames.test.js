import { createSpinner } from '../index.js'

let stdout = { out: '' }
stdout.write = symbols => {
  stdout.out += symbols
}

it('uses custom frames', () => {
  let spinner = createSpinner('', { stream: stdout })
  spinner.frames = ['.', 'o', '0', '@', '*']

  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()
  spinner.spin()

  expect(stdout.out).toMatchSnapshot()
})
