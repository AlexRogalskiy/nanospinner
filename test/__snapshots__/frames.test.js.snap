// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses custom frames 1`] = `"[1G[33m.[39m [1G[33mo[39m [1G[33m0[39m [1G[33m@[39m [1G[33m*[39m "`;
